src/main/java/FC/Frame.java

```diff
@@ -1,2 +1,212 @@
+        studyPanel.setBorder(new EmptyBorder(200,0,0,0));
+        deckName = new JLabel("Deck Name: ");
+        cardTextArea = new JLabel();    //populate it here?
+
+        correctButton = new JButton("CORRECT!");
+        incorrectButton = new JButton("INCORRECT!");
+        definitionButton = new JButton("View Definition");
+        resetButton = new JButton("RESET");
+        studyButtonPanel.add(correctButton);
+        studyButtonPanel.add(incorrectButton);
+        studyButtonPanel.add(definitionButton);
+        studyButtonPanel.add(resetButton);
+
+        studyPanel.add(cardTextArea);
+        studyPanel.add(Box.createVerticalStrut(15));
+        studyPanel.add(studyButtonPanel);
+        studyPanel.setVisible(false);
+        middlePanel.add(studyPanel);
+    }
+
+    private void setupNewDeckField() {
+        newDeckPanel = new JPanel();
+        newDeckPanel.setLayout(new BoxLayout(newDeckPanel, BoxLayout.Y_AXIS));
+        newDeckPanel.setBorder(new EmptyBorder(200,0,0,0));
+
+        newDeckLabel = new JLabel("Please Enter The Name Of Your New Deck");
+        deckNameTb = new JTextField();
+        enterDeckButton = new JButton("Enter New Deck!");
+        newDeckPanel.add(newDeckLabel);
+        newDeckPanel.add(Box.createVerticalStrut(15));
+        newDeckPanel.add(deckNameTb);
+        newDeckPanel.add(Box.createVerticalStrut(15));
+        newDeckPanel.add(enterDeckButton);
+        newDeckPanel.setVisible(false);
+        middlePanel.add(newDeckPanel);
+    }
+
+    private void newDeckClicked(){
+        newDeckPanel.setVisible(true);
+        existingDeckPanel.setVisible(false);
+        studyPanel.setVisible(false);
+    }
+    private void comboboxClicked(){
+        deckSelected = deckBox.getSelectedItem().toString();
+        newDeckPanel.setVisible(false);
+        existingDeckPanel.setVisible(true);
+        studyPanel.setVisible(false);
+
+    }
+
+
+    private void studyOrResetClicked() throws SQLException {
+        studyController = new StudyController(deckSelected);
+        studyController.startNewStudySession(deckSelected);
+        currentCard = studyController.getNextToStudy();
+        cardTextArea.setText(currentCard.getTerm());
+
+        correctButton.setEnabled(true);
+        incorrectButton.setEnabled(true);
+        definitionButton.setEnabled(true);
+        definitionButton.addActionListener(actionEvent -> cardTextArea.setText(currentCard.getDef()));
+        correctButton.addActionListener(actionEvent -> correctButtonClicked());
+        incorrectButton.addActionListener(actionEvent -> incorrectButtonClicked());
+        studyPanel.setVisible(true);
+    }
+
+    private void incorrectButtonClicked() {
+        currentCard = studyController.getNextToStudy();
+        cardTextArea.setText(currentCard.getTerm());
+    }
+
+    private void correctButtonClicked() {
+        currentCard = studyController.getNextToStudy();
```

Comment:
should only enter correctButtonClicked() once correct button is clicked - this works the first time around. **Bug** - Once user reaches end of cards and "reset" or "study" is clicked, it goes through this method, and then re-enters the method as if the correct button was clicked again.